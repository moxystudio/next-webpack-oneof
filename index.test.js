'use strict';

const alwaysOneOf = require('./index');

const createConfig = (loaderRules = []) => ({
    module: {
        rules: [...loaderRules],
    },
});

it('should return in .oneOf rules sent in .rules', () => {
    const rules = [{
        should: 'work',
    }];

    const createOneOf = (loaderRules = []) => ({
        module: {
            rules: [{
                oneOf:
                    [...loaderRules],
            }],
        },
    });

    const webpackConfig = alwaysOneOf().webpack(createConfig(rules));

    expect(webpackConfig).toMatchSnapshot();
    expect(webpackConfig).toStrictEqual(createOneOf(rules));
});

it('should call nextConfig webpack if defined', () => {
    const nextConfig = {
        webpack: jest.fn(() => 'foo'),
    };

    const webpackConfig = alwaysOneOf(nextConfig).webpack(createConfig());

    expect(nextConfig.webpack).toHaveBeenCalledTimes(1);
    expect(webpackConfig).toBe('foo');
});
