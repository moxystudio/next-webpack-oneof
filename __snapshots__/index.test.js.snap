// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add condition to Next.js built-in oneOf rule that adds JS & CSS support 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "oneOf": Array [
          Object {
            "oneOf": Array [
              Object {
                "test": /\\\\\\.\\(tsx\\|ts\\|js\\|cjs\\|mjs\\|jsx\\)\\$/,
              },
              Object {
                "test": /\\\\\\.module\\\\\\.css\\$/,
              },
            ],
            "test": /\\\\\\.\\(tsx\\|ts\\|js\\|cjs\\|mjs\\|jsx\\|css\\|scss\\|sass\\)\\$/,
          },
        ],
      },
    ],
  },
}
`;

exports[`should add noop SVG rule at the root if a SVG rule was detected 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\\\\\.svg\\$/,
      },
      Object {
        "oneOf": Array [
          Object {
            "loader": "foo",
            "test": /\\\\\\.svg\\$/,
          },
        ],
      },
    ],
  },
}
`;

exports[`should not wrap rule JS rules that should be fallthrough 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "resolve": Object {
          "fullySpecified": false,
        },
        "test": /\\\\\\.m\\?js/,
      },
      Object {
        "issuerLayer": "api",
        "test": /\\\\\\.\\(js\\|cjs\\|mjs\\)\\$/,
      },
      Object {
        "oneOf": Array [
          Object {
            "oneOf": Array [
              Object {
                "test": /\\\\\\.\\(tsx\\|ts\\|js\\|cjs\\|mjs\\|jsx\\)\\$/,
              },
              Object {
                "test": /\\\\\\.module\\\\\\.css\\$/,
              },
            ],
            "test": /\\\\\\.\\(tsx\\|ts\\|js\\|cjs\\|mjs\\|jsx\\|css\\|scss\\|sass\\)\\$/,
          },
        ],
      },
    ],
  },
}
`;

exports[`should return in .oneOf rules sent in .rules 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "oneOf": Array [
          Object {
            "test": /\\\\\\.js\\$/,
          },
        ],
      },
    ],
  },
}
`;
